File: CS/EFCore/ExportReportEF/ExportXAFReport/MyReportDataSourceHelper.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.ReportsV2;
using DevExpress.Persistent.Base.ReportsV2;

namespace ExportXafReport
{
    public class MyReportDataSourceHelper : ReportDataSourceHelper
    {
        IObjectSpaceProvider objectSpaceProvider;
        public MyReportDataSourceHelper(IObjectSpaceProvider objectSpaceProvider)
            : base(null)
        {
            this.objectSpaceProvider = objectSpaceProvider;
        }
        protected override IReportObjectSpaceProvider CreateReportObjectSpaceProvider()
        {
            return new MyReportObjectSpaceProvider(objectSpaceProvider);
        }
    }
}

File: VB/ExportReportDemoEF/ExportXafReport/MyReportDataSourceHelper.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.ReportsV2
Imports DevExpress.Persistent.Base.ReportsV2

Namespace ExportXafReport

    Public Class MyReportDataSourceHelper
        Inherits ReportDataSourceHelper

        Private objectSpaceProvider As IObjectSpaceProvider

        Public Sub New(ByVal objectSpaceProvider As IObjectSpaceProvider)
            MyBase.New(Nothing)
            Me.objectSpaceProvider = objectSpaceProvider
        End Sub

        Protected Overrides Function CreateReportObjectSpaceProvider() As IReportObjectSpaceProvider
            Return New MyReportObjectSpaceProvider(objectSpaceProvider)
        End Function
    End Class
End Namespace
